### PR TITLE
factor: Get rid of the custom `OverflowingAdd` trait

### PR DESCRIPTION
**Pull request by nbraud**

### CODE REVIEW

File: src/uu/factor/Cargo.toml [hunk 1]
```diff
@@ -12,11 +12,11 @@ categories = ["command-line-utilities"]
 edition = "2018"
 
 [build-dependencies]
-num-traits = "0.2" # used in src/numerics.rs, which is included by build.rs
+num-traits = "0.2.13" # used in src/numerics.rs, which is included by build.rs
 
 
 [dependencies]
-num-traits = "0.2"
+num-traits = "0.2.13"
```
Comment (sylvestre) [inline src/uu/factor/Cargo.toml:19]: because a missing feature has been added in 0.2.13?

Comment (nbraud) [reply src/uu/factor/Cargo.toml:19]: Yes, `OverflowingAdd` was added in 0.2.13

Comment (sylvestre) [inline src/uu/factor/Cargo.toml:19]:
```suggestion
# Needs at least version 0.2.13 for "OverflowingAdd"
num-traits = "0.2.13"
```

Comment (nbraud) [reply src/uu/factor/Cargo.toml]: Done

**Comment by nbraud** [comment]
I added a small change so that `num_traits` is only used directly in `numeric::traits`.

File: src/uu/factor/src/numeric/traits.rs [hunk 2]
```diff
@@ -6,9 +6,12 @@
 // * For the full copyright and license information, please view the LICENSE file
 // * that was distributed with this source code.
 
+pub(crate) use num_traits::{
+    identities::{One, Zero},
```
Comment (sylvestre) [inline src/uu/factor/src/numeric/traits.rs:10]: clippy is complaining:
```
WARNING: `cargo clippy`: unused imports: `One`, `Zero`
```

Comment (nbraud) [reply src/uu/factor/src/numeric/traits.rs:10]: >_>'

Comment (nbraud) [reply src/uu/factor/src/numeric/traits.rs:10]: This was again the thing with `traits` being pulled into `build.rs`.  Fixed now.

**Pull request merged**